articles/job-scheduling-in-node.js/index.md

```diff
@@ -0,0 +1,146 @@
+Now that we have created our node application and initialized it. Let's go ahead and create the server that we will use for scheduling tasks. First, type the code below to import express module and initialize it
+
+```Javascript
+const express = require('express');
+const app = express();
+```
+
+Next up, set the server to listen to port 2400. **Note: You can use a port number of your choice**
+
+```Javascript
+app.listen(2400, () => {console.log("Server started at port 2400")});
+```
+
+That's all we need for our server now. To start the server type 'node <entry point file>' on the terminal as shown
+
+```bash
+node index
+```
+
+This is the expected output from the console
+```
+Server started at port 2400
+```
+
+### Step 3 — Scheduling a Task
+Now that we have our server up and running. Let's see how to schedule a simple task before we get into sending our emails. Let's start by importing the node-cron module
+
+```Javascript
+const cron = require('node-cron');
+```
+
+To schedule a task, we pass in our *crontab* expression and the function we want to schedule on a schedule method. In this example, we  pass the default `cron` expression which will schedule the task to run every minute.
+
+```Javascript
+cron.schedule('* * * * *', () => {console.log("Task is running every minute")});
+```
+The console output is as shown
+
+```bash
+$ node index
+Server started at port 2400
+Task is running every minute
+Task is running every minute
+Task is running every minute
+...
+```
+### Step 4 — Creating The Email Function
+Let's create the function that will send emails.  First, import the node-mailer module
+
+```Javascript
+const mailer = require('nodemailer');
+```
+
+Then create a function and give it a name of your choice.
+In the function, we will create a transporter through nodemailer's `createTransport` method. We will then use the `sendMail` method to send the email.
+
+```Javascript
+function sendEmail(message){
+    // Creating a transporter
+    const transporter = mailer.createTransport({
+        host: 'smtp.ethereal.email',
+        port: 587,
+        auth: {
+            user: 'your-username',
+            pass: 'your-password'
+        }
+    });
+    //sending the email
+    transporter.sendMail({
+        from: '"Peter" <peter@kayere.com>',
+        to: '"You there" <you@there.com>',
+        subject: 'Scheduled Email',
+        text: message
+    })
+        .then(_ => {console.log("Email has been sent")})
+        .catch(error => {console.log(error)});
+}
+```
+
+### Step 5 — Scheduling The Email
+In this step, we are going to use the schedule method we created above to schedule the `sendEmail` function. We will schedule the function to run at midnight every Sunday. To represent this using the *crontab* syntax, we will pass zeros for the minutes and hours asterisk and Sunday for the day. This is how we will do it.
+
+```Javascript
+cron.schedule('0 0 * * sunday', sendEmail("Hey there, this email was sent to you automatically"));
```

Comment:
You can set this function to run on a shorter time span. Something like every 5 mins to show readers a better example.